editor/editor_file_system.cpp

```diff
@@ -1578,6 +1586,45 @@ String EditorFileSystem::_get_global_script_class(const String &p_type, const St
 	return String();
 }
 
+void EditorFileSystem::_update_file_icon_path(EditorFileSystemDirectory::FileInfo *file_info) {
+	String icon_path;
+	if (file_info->script_class_icon_path.is_empty()) {
+		if (!file_info->deps.is_empty()) {
```

Comment:
You can reduce nesting now.
```suggestion
	if (file_info->script_class_icon_path.is_empty() && !file_info->deps.is_empty()) {
```

Reply:
Done :)